cmd/virsnap/list.go

```diff
@@ -90,45 +90,50 @@ func listRun(cmd *cobra.Command, args []string) {
 			continue
 		}
 
-		defer virt.FreeSnapshots(logger, snapshots)
-
-		// print the VM header to stdout
-		fmt.Printf("%s (current state: %s, %d snapshots total)\n",
-			color.BGreen(vm.Descriptor.Name), vmstate,
-			len(snapshots))
-
-		// print no snapshot table if there are no snapshots for this VM
-		if len(snapshots) == 0 {
-			continue
-		}
+		// anonymous function for safely calling defer
+		func() {
```

Comment:
I don't really understand why this has to be an anonymous function. The function is returning after this whole block anyway, so there shouldn't be any issues with calling ?